modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java

```diff
@@ -456,7 +456,8 @@ public String toVarName(String name) {
 
     @Override
     public String toParamName(String name) {
-        return toVarName(name);
+        // $ref appears to be all uppercase which is contrary to rustfmt practice so lowercase parameters
+        return toVarName(name.toLowerCase(Locale.ROOT));
     }
```

Comment:
CI samples check reported the following:
```
diff --git a/samples/client/petstore/rust/hyper/petstore/docs/PetApi.md b/samples/client/petstore/rust/hyper/petstore/docs/PetApi.md
index 7deea32c..b94b34a8 100644
--- a/samples/client/petstore/rust/hyper/petstore/docs/PetApi.md
+++ b/samples/client/petstore/rust/hyper/petstore/docs/PetApi.md
@@ -47,7 +47,7 @@ Name | Type | Description  | Required | Notes
 
 ## delete_pet
 
-> delete_pet(pet_id, api_key)
+> delete_pet(petid, api_key)
 Deletes a pet
 
 
@@ -57,7 +57,7 @@ Deletes a pet
 
 Name | Type | Description  | Required | Notes
 ------------- | ------------- | ------------- | ------------- | -------------
-**pet_id** | **i64** | Pet id to delete | [required] |
+**petid** | **i64** | Pet id to delete | [required] |
 **api_key** | Option<**String**> |  |  |
```

Can we remove this change for the time being as it's a breaking change? 

I think underscore should be allowed, right?

Reply:
Hmm... maybe instead of lower casing it should be converting to camel case?  Would solve the issue referenced in the comment while preserving the eventual snake case.

Reply:
Looking around a little and it appears [WordUtils](https://javadoc.io/static/org.apache.commons/commons-lang3/3.12.0/org/apache/commons/lang3/text/WordUtils.html) might have the necessary method.

```
return toVarName(WordUtils.uncapitalize(WordUtils.capitalizeFully(name)));
``` 
would likely suffice.  Does this sound correct (I'm not sure of all the use cases)?  If so, I can add an update to the PR.

Reply:
I ran https://github.com/rust-lang/rust-clippy but didn't see any warnings related to using underscore in the variable names.

Reply:
Underscores are expected. I'm assuming that the Java parser does camelcase and the Rust generator converts to snakecase at some point.

The original issue is that $ref is all uppercase which clippy will complain about.

Reply:
I may have missed it. Do you have a sample spec to reproduce the "original issue"?

Reply:
>         // $ref appears to be all uppercase 

I don't think that's a correct statement. Please have a look at petstore.yaml.

Reply:
yep... not sure the use case where I was seeing that.  I've reverted the change.  Also found that some clippy allows are still necessary so reverted that as well.  fwiw, I'm not using the generated crate but instead embed the modules into a workspace using a `build.rs` script.  Would be a nice feature if one could choose to just generate the various modules.